File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int i : tangerine) {
            map.put(i,map.getOrDefault(i,0)+1);
        }
        for(int key:map.keySet()) {
            pq.add(map.get(key));
        }
        while(k>0) {
            k-=pq.poll();
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const arr = new Array(10000001).fill(0) 
    for(let i of tangerine) { 
        arr[i] += 1
    }
    arr.sort((a,b)=> b-a)
    for(let el of arr) {
        k-=el
        answer++
        if(k<=0) return answer
    }
    return answer;
}

//서로 다른 종류가 최소
//같은 종류를 많이 담아라